core/src/com/unciv/ui/popups/options/AdvancedTab.kt

```diff
@@ -98,29 +100,50 @@ class AdvancedTab(
             optionsPopup.reopenAfterDisplayLayoutChange()
         }
     }
-    
-    private fun addMaxAutosavesStored() {
+
+    private fun addAutosaveField() {
         add("Number of autosave files stored".toLabel()).left().fillX()
-        
-        val maxAutosavesStoredSelectBox = SelectBox<Int>(skin)
-        val maxAutosavesStoredArray = Array<Int>()
-        maxAutosavesStoredArray.addAll(1,2,5,10,15,20,35,50,100,150,200,250)
-        maxAutosavesStoredSelectBox.items = maxAutosavesStoredArray
-        maxAutosavesStoredSelectBox.selected = settings.maxAutosavesStored
-        
-        add(maxAutosavesStoredSelectBox).pad(10f).row()
-        
-        maxAutosavesStoredSelectBox.onChange {
-            settings.maxAutosavesStored = maxAutosavesStoredSelectBox.selected
-        }
+        val autosaveFieldTable = Table()
+        val autoSaveTrunsTextField = UncivTextField("",settings.maxAutosavesStored.toString())
+        autoSaveTrunsTextField.setTextFieldFilter { _, c -> c in " 1234567890" }
+        autosaveFieldTable.add(autoSaveTrunsTextField)
+        val autoSaveTrunsTextFieldButton = "Enter".toTextButton()
+
+        autosaveFieldTable.add(
+            autoSaveTrunsTextFieldButton.onClick {
```

Comment:
Onclick and "add to table" should be 2 different lines

button.onclick{...}
autosaveFieldTable.add(button)